client/api/src/client.rs

```diff
@@ -307,6 +307,8 @@ pub struct FinalityNotification<Block: BlockT> {
 	/// Finalized block header.
 	pub header: Block::Header,
 	/// Path from the old finalized to new finalized parent (implicitly finalized blocks).
+	///
+	/// This maps to the range `[old_finalized, new_finalized]`.
```

Comment:
```suggestion
	/// This maps to the range `[old_finalized, new_finalized)`.
```
The new_finalized hash is not included in the `FinalityNotification::tree_route`. Is extracted [here](https://github.com/paritytech/substrate/blob/5597a93a8c8b1ab578693c68549e3ce1902f3eaf/client/api/src/client.rs#L335) when mapping the `FinalizeSummary` to `FinalityNotification`.

Reply:
It already contained the `new_finalized`, so this suggestion is not really correct.